### DISCUSSION TITLE
OpenAI API seemed not under proxy using tts

### DISCUSSION THREAD

**Discussion by nijisakai** [open-webui/open-webui#1357] [General]
# Bug Report

## Description

**Bug Summary:**
when add -e OPENAI_API_BASE_URLS="https://api.openai.com/v1 i got a blank screen and error says after a while: no connection to api.openai.com. remove this environment and everything goes fine.
Also, although i set proxy and i can use gpt-3.5 from litellm, i still can't use Audio from OpenAI API, gives a 500 error.

## Environment

```
docker run --network=host -v open-webui:/app/backend/data -e OLLAMA_BASE_
URLS="http://127.0.0.1:11434;http://172.1.1.4:11434" -e http_proxy=http://172.2.2.4:7890 -e https_proxy=http://172.2.2.4:7890 -e no_proxy=127.0.0.1,localhost --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

**Comment by mbotsu** [comment]
Hi.

How about this method?
`aiohttp.ClientSession` can use environment variable proxies by setting `trust_env=True` as an argument.

```
diff --git a/backend/apps/openai/main.py b/backend/apps/openai/main.py
index 85ee531f..68717c19 100644
--- a/backend/apps/openai/main.py
+++ b/backend/apps/openai/main.py
@@ -187,7 +187,7 @@ async def fetch_url(url, key):
     try:
         if key != "":
             headers = {"Authorization": f"Bearer {key}"}
-            async with aiohttp.ClientSession(timeout=timeout) as session:
+            async with aiohttp.ClientSession(timeout=timeout, trust_env=True) as session:
                 async with session.get(url, headers=headers) as response:
                     return await response.json()
         else:
```

Now mount the source in `docker-compose.yml`

```
volumes:
  - ./open-webui/backend/apps:/app/backend/apps
```

Thank.

**Reply by nijisakai** [reply]
Thanks. In Ver.0.1.124 i think the problem has gone.

**Reply by mbotsu** [reply]
Thank you for your reply.

The environment I evaluated is v0.1.125.
Thanks.
